src/XMakeBuildEngine/UnitTestsPublicOM/Construction/ProjectFormatting_Tests.cs

```diff
@@ -0,0 +1,462 @@
+﻿using Microsoft.Build.Construction;
+using Microsoft.Build.Evaluation;
+using Microsoft.Build.UnitTests;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Xml;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.Build.Engine.OM.UnitTests.Construction
+{
+    public class ProjectFormatting_Tests : IDisposable
+    {
+        private readonly ITestOutputHelper _testOutput;
+
+        public ProjectFormatting_Tests(ITestOutputHelper testOutput)
+        {
+            _testOutput = testOutput;
+            Setup();
+        }
+
+        /// <summary>
+        /// Clear out the cache
+        /// </summary>
+        public void Setup()
+        {
+            ProjectCollection.GlobalProjectCollection.UnloadAllProjects();
+            GC.Collect();
+        }
+
+        /// <summary>
+        /// Clear out the cache
+        /// </summary>
+        public void Dispose()
+        {
+            Setup();
+        }
+
+        [Fact]
+        public void ProjectCommentFormatting()
+        {
+            string content = ObjectModelHelpers.CleanupFileContents(@"
+<Project DefaultTargets=`Build` ToolsVersion=`msbuilddefaulttoolsversion` xmlns=`msbuildnamespace`>
+  <ItemGroup>
+    <ProjectReference Include=`..\CLREXE\CLREXE.vcxproj`><!-- Comment -->
+      <Project>{3699f81b-2d03-46c5-abd7-e88a4c946f28}</Project>
+    </ProjectReference>
+  </ItemGroup>
+</Project>");
+
+            string reformattedContent = ObjectModelHelpers.CleanupFileContents(@"
+<Project DefaultTargets=`Build` ToolsVersion=`msbuilddefaulttoolsversion` xmlns=`msbuildnamespace`>
+  <ItemGroup>
+    <ProjectReference Include=`..\CLREXE\CLREXE.vcxproj`>
+      <!-- Comment -->
+      <Project>{3699f81b-2d03-46c5-abd7-e88a4c946f28}</Project>
```

Comment:
I'm not sure I prefer this behavior. It can be useful to have both
```C
// Long comments explaining
code;
```

and
```C
code; // clarified
```